src/condor_tests/test_htcondor_cli_sys_status.py

```diff
@@ -0,0 +1,183 @@
+        "ExpectedDaemon" : "MASTER",
+        "ExpectedHP" : 63,
+    },
+    # Verify 0.97 rdcdc -> 31 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "DaemonMaster",
+        "RecentDaemonCoreDutyCycle" : 0.97,
+        "ExpectedDaemon" : "MASTER",
+        "ExpectedHP" : 31,
+    },
+    # Verify MyType Schedd and 0 rdcdc -> 100 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 100,
+    },
+    # Verify Schedd 1 rdcdc -> 28 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 1,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 28,
+    },
+    # Verify Schedd with 0 bytes per second Upload -> 93
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "TransferQueueNumUploading" : 1,
+        "FileTransferUploadBytesPerSecond_1m" : 0,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 93,
+    },
+    # Verify Download MB Waiting < 100 -> 100 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "FileTransferMBWaitingToDownload" : 10,
+        "TransferQueueDownloadWaitTime" : 100000,
+        "TransferQueueNumWaitingToDownload" : 3,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 100,
+    },
+    # Verify Download MB Waiting > 100 && Wait time < 5m -> 100 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "FileTransferMBWaitingToDownload" : 250,
+        "TransferQueueDownloadWaitTime" : 20,
+        "TransferQueueNumWaitingToDownload" : 3,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 100,
+    },
+    # Verify Download MB Waiting > 100 && Wait time > 5m -> 93 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "FileTransferMBWaitingToDownload" : 250,
+        "TransferQueueDownloadWaitTime" : 100000,
+        "TransferQueueNumWaitingToDownload" : 3,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 93,
+    },
+]
+
+@action(params={name: name for name in TEST_CASES})
+def run_commands(default_condor, request):
+    with default_condor.use_config():
+        # HTCondor cli logger outputs to stderr?
+        p = subprocess.run(TEST_CASES[request.param], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=60)
+        output = p.stdout.rstrip().decode() + p.stderr.rstrip().decode()
+        yield output
```

Comment:
Perhaps I have should have looked at the implementation, but did you miss Ornithology's `run_command`?  If that doesn't do what you want, should we consider amending it?

Reply:
I totally forgot about ornithologies built in `run_command`... I will look into it